### PR TITLE
Clean descriptor for Mono's SPC

### PR DESCRIPTION
**Pull request by marek-safar**
- Remove names pointing to non-existent members
- Fix spaces in method signatures which break matching rules
- Remove duplicate entries
- Format file for consistency

### CODE REVIEW

File: src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml [hunk 1]
```diff
@@ -757,9 +750,5 @@
 		<type fullname="System.Threading.ThreadStartException">
 			<method name=".ctor" />
 		</type>
-		<type fullname="System.Runtime.InteropServices.Marshal">
-			<!-- marshal-ilgen.c: emit_marshal_custom_get_instance -->
-			<method name="GetCustomMarshalerInstance" />
```
Comment (akoeplinger) [inline src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml:762]: this is used too:

https://github.com/dotnet/runtime/blob/fbe4d10539a60f3a536ade5cdce991ae09fa9b84/src/mono/mono/metadata/marshal-ilgen.c#L4479

Comment (vargaz) [reply src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml:762]: This was added above to the existing Marshal type block.

File: src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml [hunk 2]
```diff
@@ -662,19 +657,19 @@
...
 		</type>
-		
+
 		<!-- domain.c: mono_defaults.safehandle_class -->
 		<type fullname="System.Runtime.InteropServices.SafeHandle" preserve="fields">
 			<!-- marshal.c (init_safe_handle) -->
 			<method name="DangerousAddRef" />
 			<method name="DangerousRelease" />
-			<!-- marshal-ilgen.c (emit_marshal_safehandle_ilgen) -->
-			<method signature="System.Void .ctor()" />
```
[the remark this answers is not in the document]
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml:675]: It's not, this is looking for ctor on derived types from SafeHandle

File: src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml (hunk 1 above)
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/ILLink/System.Private.CoreLib.xml:762]: Yep, it's still there https://github.com/dotnet/runtime/pull/38442/files#diff-621abfb605caafcaac53649d9dde3cc0R661

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**